src/setup_payload/QRCodeSetupPayloadGenerator.cpp

```diff
@@ -138,7 +138,7 @@ CHIP_ERROR QRCodeSetupPayloadGenerator::generateTLVFromOptionalData(SetupPayload
 
     TLV::TLVWriter rootWriter;
     rootWriter.Init(tlvDataStart, maxLen);
-    rootWriter.ImplicitProfileId = chip::Protocols::kProtocol_ServiceProvisioning;
+    rootWriter.ImplicitProfileId = chip::Protocols::kProtocol_SetupPayload;
```

Comment:
This doesn't match the spec.  The spec does not mention any use of profile-specific tags here or an implicit profile.  As far as I can tell, in the spec:

1. The outermost struct is required.
2. The outermost struct is tagged with an anonymous tag.
3. The things inside the struct use context tags.  The code seems to have that part right, when it outputs the struct.

Importantly, the spec does not define a "setup payload protocol" that I can tell.

Reply:
This is meant to be provisional until the NetworkProvisioningCluster comes in.

Reply:
I don't see how the Network Provisioning Cluster affects this... but if this is temporary code, why are we making this change to it in the first place?

Reply:
Ah. Seems like the spec has been changed by: `https://github.com/CHIP-Specifications/connectedhomeip-spec/commit/2ae014a26dba265e96754f1daade4599356355fc`

There are more changes needed to accommodate for the new spec. If will fix it here.

Reply:
I changed the code to remove the optimisation for including/excluding the top level container based on the size of the data to be encoded. And then, I also remove the usage of `ImplicitProfileId` since the updated spec does not use that anymore.

Reply:
@bzbarsky-apple Re-requesting review using a comment. The GitHub UI seems weird at the moment to me.